airflow/auth/managers/models/base_user.py

```diff
@@ -37,6 +38,11 @@ def is_active(self) -> bool:
     def is_anonymous(self) -> bool:
         ...
 
+    @property
+    @abstractmethod
+    def perms(self) -> Any:
```

Comment:
can't we be more specific ? It seems we use that in a `x in user.perms` expression, so can't we type this as `list | set` maybe ?

Reply:
Hard to say, it is very hard to predict all use cases from all auth managers. That's why I prefer to stay on the safe side. Anytime we use `x in user.perms` is in the FAB auth manager, thus we know the structure of `perms`. But across auth managers, this border can be very blurry